torch/distributed/tensor/debug/visualize_sharding_example.py

```diff
@@ -0,0 +1,77 @@
+# OMP_NUM_THREADS=1 torchrun --nproc_per_node=4 visualize_sharding_example.py
+import contextlib
+import importlib.util
+import os
+
```

Comment:
for this example file, i'm fine landing it here for now, but i think longer term this might make sense to move the content into a 'tutorial' page on pytorch tutorials repo.

Reply:
Agreed.

Reply:
some duplicate files already in the examples folder, I think we should put it there first https://github.com/pytorch/pytorch/pull/152027#discussion_r2057401742